drivers/include/epd_bw_spi.h

```diff
@@ -0,0 +1,273 @@
+ * Various display controllers are currently supported out of the box, see @ref epd_bw_spi_params.h.
+ * Please open an issue or pull request with your controller details (size, Vcom, LUTs)
+ * if your display controller is not included yet.
+ *
+ * Use of this driver requires knowing the parameters of your display.
+ * See epd_bw_spi_params_t and @ref epd_bw_spi_params.h for more details on the parameters.
+ * Note that while the reset and busy pins are optional, using them is highly recommended.
+ *
+ * This driver has @ref drivers_disp_dev support.
+ *
+ * @{
+ * @file
+ * @brief       Generic black/white e-paper/e-ink display SPI driver.
+ *
+ * @author      Silke Hofstra <silke@slxh.eu>
+ */
+
+#ifndef EPD_BW_SPI_H
+#define EPD_BW_SPI_H
+
+#include "periph/spi.h"
+#include "periph/gpio.h"
+
+#ifdef MODULE_DISP_DEV
+#include "disp_dev.h"
+#endif
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#define EPD_BW_SPI_COLOR_WHITE (0xFF)   /**< White (8x1 pixels) */
+#define EPD_BW_SPI_COLOR_BLACK (0x00)   /**< Black (8x1 pixels) */
+
+/**
+ * @brief   Data entry mode settings.
+ *
+ * This setting affect the automatic increment/decrement of the address counters.
+ */
+typedef enum {
+    EPD_BW_SPI_Y_DEC_X_DEC  = 0x0, /**< Y decrement, X decrement */
+    EPD_BW_SPI_Y_DEC_X_INC  = 0x1, /**< Y decrement, X increment */
+    EPD_BW_SPI_Y_INC_X_DEC  = 0x2, /**< Y increment, X decrement */
+    EPD_BW_SPI_Y_INC_X_INC  = 0x3, /**< Y increment, X increment */
+} epd_bw_spi_entry_mode_t;
+
+/**
+ * @brief   Additional status codes for black/white SPI e-paper displays.
+ */
+enum {
+    EPD_BW_SPI_DC_FAIL      = -5,
+    EPD_BW_SPI_RST_FAIL     = -6,
+    EPD_BW_SPI_BUSY_FAIL    = -7,
+};
+
+/**
+ * @brief   Display controller parameters.
+ */
+typedef struct {
+    uint8_t vcom;               /**< VCOM voltage level */
+    const uint16_t size_x;      /**< supported number of horizontal pixels */
+    const uint16_t size_y;      /**< supported number of vertical pixels */
+    const uint8_t lut_size;     /**< size of the waveform lookup table */
+    const uint8_t *lut_full;    /**< lookup table for a full display refresh */
+    const uint8_t *lut_part;    /**< lookup table for a partial display refresh */
+} epd_bw_spi_controller_t;
+
+/**
+ * @brief   SPI display device initialisation parameters.
+ */
+typedef struct {
+    spi_t spi;                              /**< SPI device that the display is connected to */
+    spi_clk_t spi_clk;                      /**< SPI clock speed to use */
+    gpio_t cs_pin;                          /**< pin connected to the CHIP SELECT line */
+    gpio_t dc_pin;                          /**< pin connected to the DC line */
+    gpio_t rst_pin;                         /**< pin connected to the reset line (optional) */
+    gpio_t busy_pin;                        /**< pin connected to the busy line (optional) */
+    bool busy_value;                        /**< expected value for the busy pin */
```

Comment:
Another documentation nitpick:

```suggestion
    bool busy_value;                        /**< expected value for the busy pin when the display is busy */
```